raster/src/main/scala/geotrellis/raster/FeatureExtraction.scala

```diff
@@ -0,0 +1,58 @@
+package geotrellis.raster
+
+import geotrellis.raster.rasterize.Rasterizer
+import geotrellis.vector.{Feature, Geometry, Point, PointFeature}
+import spire.syntax.cfor._
+
+import scala.collection.mutable.ListBuffer
+
+trait FeatureExtraction[G <: Geometry, T <: CellGrid[Int], D] {
+  def features(geom: G, raster: Raster[T]): Array[Array[PointFeature[D]]]
```

Comment:
@echeipesh the main question here, is this interface smth that we want? 

Also a quick notion about bands subsetting - bands subset operation should be done as a step before applying this funtion and it definitely should not be a part of this interface.

The output `Array` is for `bands`, the input `Array` is for `PointFeatures`. Do we need an interface that will allow to implement any `Features` 'extraction'?